File: implementations/wrap-assemblyscript/src/__tests__/cases/001-module-functions/output/Module/module.ts
import * as Types from "..";

import {
  BigInt,
  BigNumber,
  Box,
  JSON,
} from "@polywrap/wasm-as";

export abstract class ModuleBase {
  abstract function1(
    args: Types.Args_function1
  ): string;

  abstract function2(
    args: Types.Args_function2
  ): string | null;
}

File: implementations/wrap-rust/src/__tests__/cases/001-module-functions/output/module/module.rs
use polywrap_wasm_rs::{
    BigInt,
    BigNumber,
    Map,
    JSON,
    JSONString,
    BigIntWrapper,
    ByteBuf
};
use crate::{
    ArgsFunction1,
    ArgsFunction2,
};

pub struct Module;

pub trait ModuleTrait {
  fn function1(args: ArgsFunction1) -> Result<String, String>;

  fn function2(args: ArgsFunction2) -> Result<Option<String>, String>;
}
